klippa/src/fvar.rs

```diff
@@ -0,0 +1,54 @@
+//! impl subset() for fvar table
+
+use crate::{NameidClosure, Plan};
+use write_fonts::read::{tables::fvar::Fvar, types::NameId};
+
+impl<'a> NameidClosure for Fvar<'a> {
+    //TODO: support partial-instancing
+    fn collect_name_ids(&self, plan: &mut Plan) {
+        let Ok(axis_instance_array) = self.axis_instance_arrays() else {
+            return;
+        };
+        plan.name_ids
+            .extend_unsorted(axis_instance_array.axes().iter().map(|x| x.axis_name_id()));
+
+        for instance_record in axis_instance_array
+            .instances()
+            .iter()
+            .filter_map(|x| x.ok())
+        {
+            plan.name_ids.insert(instance_record.subfamily_name_id);
+            if let Some(name_id) = instance_record.post_script_name_id {
+                if name_id != NameId::INVALID {
+                    plan.name_ids.insert(name_id);
+                }
+            }
```

Comment:
Since the spec does allow these to be sparse, I think we should handle it in `InstanceRecord::read_with_args()` rather than here, returning `None` when the PostScript name id is `0xFFFF`.

Specifically, here: https://github.com/googlefonts/fontations/blob/5e0b6ae79bbfb98afaa35173a5752e8ec6d2a71c/read-fonts/src/tables/instance_record.rs#L52

I'd change that code to something like:
```rust
let id: NameId = cursor.read()?;
// From <https://learn.microsoft.com/en-us/typography/opentype/spec/fvar#instancerecord>:
// "If the value is 0xFFFF, then the value is ignored, and no
// PostScript name equivalent is provided for the instance."
(id.to_u16() != 0xFFFF).then_some(id)
```

And then we can avoid the invalid check (and remove the INVALID const in NameId).

Reply:
oh yes this makes more sense. Thanks!